onebusaway-gtfs-transformer/src/main/java/org/onebusaway/gtfs_transformer/impl/AnomalyCheckFutureTripCounts.java

```diff
@@ -0,0 +1,314 @@
+    private String dayAvgTripMapUrl;
+
+    @CsvField(optional = true)
+    private String dayAvgTripMapFile;
+
+    @CsvField(optional = true)
+    private double percentageMatch = 10;
+
+    @CsvField(optional = true)
+    private boolean silentMode = false;
+
+
+    @Override
+    public String getName() {
+        return this.getClass().getSimpleName();
+    }
+
+    @Override
+    public void run(TransformContext context, GtfsMutableRelationalDao dao) {
+
+        Collection<Date> datesToIgnore;
+        SetListener datesToIgnoreListener = new SetListener();
+        datesToIgnoreListener = (SetListener) readCsvFrom(datesToIgnoreListener,datesToIgnoreUrl, datesToIgnoreFile);
+        datesToIgnore = datesToIgnoreListener.returnContents();
+
+
+        Collection<Date> holidays;
+        SetListener holidaysListener = new SetListener();
+        holidaysListener = (SetListener) readCsvFrom(holidaysListener,holidaysUrl,holidaysFile);
+        holidays = holidaysListener.returnContents();
+
+
+        Map<String, Double> dayAvgTripsMap = new HashMap<String, Double>();
+        MapListener mapListener = new MapListener();
+        mapListener = (MapListener) readCsvFrom(mapListener,dayAvgTripMapUrl,dayAvgTripMapFile);
+        dayAvgTripsMap = mapListener.returnContents();
+
+        SimpleDateFormat dayOfWeekFormat = new SimpleDateFormat("EEEE");
+        for (Date date : holidays) {
+            _log.info("holiday: " + date.toString() + " on a " + dayOfWeekFormat.format(date));
+        }
+        for (Date date : datesToIgnore) {
+            _log.info("ignore: " + date.toString());
+        }
+        for (String key : dayAvgTripsMap.keySet()){
+            System.out.println("key: " + key + " value: " + dayAvgTripsMap.get(key));
+        }
+
+
+        ExternalServices es =  new ExternalServicesBridgeFactory().getExternalServices();
+
+        String[] days = {"Monday","Tuesday","Wednesday","Thursday","Friday","Saturday","Sunday","Holiday"};
+
+
+        Map<Date, Integer> dateTripMap = getDateTripMap(dao);
+
+
+        int dayCounter = 0;
+        Map<String, ArrayList<Double>> dayAvgTripsMapUpdate = new HashMap<String, ArrayList<Double>>();
+
+
+        for (String day:days){
+            dayAvgTripsMapUpdate.put(day, new ArrayList<Double>());
+        }
+
+        while (true) {
+            Date day = removeTime(addDays(new Date(), dayCounter));
+            if (dateTripMap.get(day) == null){
+                break;
+            }
+            int tripCount = dateTripMap.get(day);
+            if ((tripCount < dayAvgTripsMap.get(dayOfWeekFormat.format(day))*(1+percentageMatch/100)) &&
+                    (tripCount > dayAvgTripsMap.get(dayOfWeekFormat.format(day))*(1-percentageMatch/100))){
+                _log.info(day + " has " + tripCount + " trips, and that's within reasonable expections");
+                dayAvgTripsMapUpdate.get(dayOfWeekFormat.format(day)).add((double) tripCount);
+            }
```

Comment:
These if/else clauses aren't mutually exclusive.  It looks like this will skew your dayAvgTripsMapUpdate if a holiday is within bounds.